circuitbreaker/circuitbreaker.go

```diff
@@ -15,6 +16,19 @@ var ErrCircuitBreakerOpen = errors.New("circuit breaker open")
 // State of a CircuitBreaker.
 type State int
 
+func (s State) String() string {
+	switch s {
+	case ClosedState:
+		return "closed"
+	case OpenState:
+		return "open"
+	case HalfOpenState:
+		return "half-open"
+	default:
+		panic(fmt.Sprintf("invalid state %d", s))
```

Comment:
Wasn't sure what to do here. Panic seemed the most obvious way to signal that there's a problem but I can also change this to return `unknown` or something.

Reply:
Yea let's just return a string for unknown instead